pkg/virt-controller/watch/vm/vm.go

```diff
@@ -3003,12 +3003,10 @@ func (c *Controller) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineI
 	if err != nil {
 		return vm, vmi, handleSynchronizerErr(err), nil
 	}
-	if !equality.Semantic.DeepEqual(vm.Spec, syncedVM.Spec) {
-		return syncedVM, vmi, nil, nil
-	}
 
 	vm.ObjectMeta = syncedVM.ObjectMeta
 	vm.Spec = syncedVM.Spec
+	vm.Status = syncedVM.Status
```

Comment:
This seems weird, why don't we just assign the vm?

Reply:
This was done originally to just copy other third party sync functions being called here but I agree it's pointless and we should just replace vm with syncedVM.